--- a/mount_root.c
+++ b/mount_root.c
@@ -79,6 +79,8 @@ static bool is_cr881x(void)
 	for (i = 0; i + sizeof("xiaomi,cr881x") - 1 <= n; i++) {
 		if (!memcmp(buf + i, "xiaomi,cr881x", sizeof("xiaomi,cr881x") - 1))
 			return true;
+		if (!memcmp(buf + i, "redmi,ax3000", sizeof("redmi,ax3000") - 1))
+			return true;
 	}
 
 	return false;
--- a/factoryreset.c
+++ b/factoryreset.c
@@ -59,6 +59,8 @@ static bool is_cr881x(void)
 	for (i = 0; i + sizeof("xiaomi,cr881x") - 1 <= n; i++) {
 		if (!memcmp(buf + i, "xiaomi,cr881x", sizeof("xiaomi,cr881x") - 1))
 			return true;
+		if (!memcmp(buf + i, "redmi,ax3000", sizeof("redmi,ax3000") - 1))
+			return true;
 	}
 
 	return false;
